packages/@cdktf/hcl2cdk/test/helpers/convert.ts

```diff
@@ -296,42 +302,48 @@ const createTestCase =
         }, 500_000);
 
         if (
-          shouldSynth === Synth.yes ||
-          shouldSynth === Synth.yes_all_languages
+          includeSynthTests &&
```

Comment:
One alternative that could make this flag a bit more visible would be to render `it.skip` if the flag is set. That way it might be easier to discover this behavior (even if documented) as a contributor if tests fail in CI while running locally seems fine. That way Jest would at least print the skipped tests.